virtualizarr/tests/test_readers/test_dmrpp.py

```diff
@@ -1,22 +1,353 @@
+from pathlib import Path
+from xml.etree import ElementTree as ET
+
+import numpy as np
 import pytest
 import xarray as xr
+import xarray.testing as xrt
 
 from virtualizarr import open_virtual_dataset
+from virtualizarr.manifests.manifest import ChunkManifest
+from virtualizarr.readers.dmrpp import DMRParser
 from virtualizarr.tests import network
 
 urls = [
     (
-        "netcdf4",
-        "https://github.com/OPENDAP/bes/raw/3e518f6dc2f625b0b83cfb6e6fd5275e4d6dcef1/modules/dmrpp_module/data/dmrpp/chunked_threeD.h5",
-        "dmrpp",
-        "https://github.com/OPENDAP/bes/raw/3e518f6dc2f625b0b83cfb6e6fd5275e4d6dcef1/modules/dmrpp_module/data/dmrpp/chunked_threeD.h5.dmrpp",
+        "https://its-live-data.s3-us-west-2.amazonaws.com/test-space/cloud-experiments/dmrpp/20240826090000-JPL-L4_GHRSST-SSTfnd-MUR25-GLOB-v02.0-fv04.2.nc",
+        "https://its-live-data.s3-us-west-2.amazonaws.com/test-space/cloud-experiments/dmrpp/20240826090000-JPL-L4_GHRSST-SSTfnd-MUR25-GLOB-v02.0-fv04.2.nc.dmrpp",
     )
+    # TODO: later add MUR, SWOT, TEMPO and others by using kerchunk JSON to read refs (rather than reading the whole netcdf file)
```

Comment:
Yes. The current testing approach is to load a netcdf, generate the references, and then compare with the dmrpp parsed version. However this is not ideal for larger files (especially when these tests are part of GitHub Actions that run frequently). So the alternative is to generate kerchunk (or maybe icechunk?) references so only a small JSON needs to be read to compare the datasets. I have disabled this test for now due to the `fill_val` issue mentioned above

Reply:
Is there a reason the test needs to rely on kerchunk json at all? I would quite like to move towards a testing strategy for VirtualiZarr where kerchunk is not involved for anything other than explicitly testing the kerchunk reader & writer.

What might be better is to have a small netCDF, then the expected byte ranges and so on are written into the test itself, by explicitly creating `Manifest` objects. Then you compare that manually-constructed virtual dataset to the virtual dataset created by the DMR parser in `open_virtual_dataset`.

In general I think it was a mistake for me to create a testing paradigm that encouraged using kerchunk where it wasn't actually necessary.

Reply:
That approach does make sense in general. This specific test is sort of a "real-world" check to make sure the parser works on real NASA datasets and their associated dmrpp files. These files could have unique characteristics that the other tests have missed. The reason for kerchunk JSON is just so I can make references for those already existing netcdf files.

Maybe this could be an external test (not part of this repo) that is used to catch bugs and add updates to the main dmrpp test suite.

Reply:
I think we can leave this for a follow-up